api-inventory-ext.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * MinIO Go Library for Amazon S3 Compatible Cloud Storage
+ * Copyright 2015-2025 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"iter"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/minio/minio-go/v7/internal/json"
+)
+
+// This file contains the inventory API extension for MinIO server. It is not
+// compatible with AWS S3.
+
+func makeInventoryReqMetadata(bucket string, urlParams ...string) requestMetadata {
+	urlValues := make(url.Values)
+	urlValues.Set("minio-inventory", "")
+	if len(urlParams)%2 != 0 {
+		panic("invalid urlParams passed - this should not happen")
+	}
+	for i := 0; i < len(urlParams); i += 2 {
+		urlValues.Set(urlParams[i], urlParams[i+1])
+	}
+
+	return requestMetadata{
+		bucketName:  bucket,
+		queryValues: urlValues,
+	}
+}
+
+// GenerateInventoryConfigYAML - calls the inventory YAML template generation
+// endpoint.
+func (c *Client) GenerateInventoryConfigYAML(ctx context.Context, bucket, id string) (string, error) {
+	reqMeta := makeInventoryReqMetadata(bucket, "generate", "", "id", id)
+	resp, err := c.executeMethod(ctx, http.MethodGet, reqMeta)
+	defer closeResponse(resp)
+	if err != nil {
+		return "", err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return "", httpRespToErrorResponse(resp, bucket, "")
+	}
+	buf := new(strings.Builder)
+	_, err = io.Copy(buf, resp.Body)
+	return buf.String(), err
+}
+
+// PutBucketInventoryConfiguration - calls the inventory configuration
+// endpoint to create or update an inventory configuration for a bucket.
+func (c *Client) PutBucketInventoryConfiguration(ctx context.Context, bucket string, id string, yamlDef []byte) error {
```

Comment:
if you are going to return `yamlDef string` then it is better to take yamlDef as string input here as well. 

Also add some options pattern so that we can allow extensions to this API when it becomes apparent in future

Reply:
I've done this, but I can't really think of future extensions here, everything would likely happen in the yaml definition itself.